x-pack/plugins/endpoint/common/generate_data.ts

```diff
@@ -350,18 +354,44 @@ export class EndpointDocGenerator {
    * Creates an alert event and associated process ancestry. The alert event will always be the last event in the return array.
    * @param alertAncestors - number of ancestor generations to create
    */
-  public createAlertEventAncestry(alertAncestors = 3): Event[] {
+  public createAlertEventAncestry(
+    alertAncestors = 3,
+    relatedEventsPerNode = 5,
+    pctWithRelated = 30
+  ): Event[] {
     const events = [];
     const startDate = new Date().getTime();
     const root = this.generateEvent({ timestamp: startDate + 1000 });
     events.push(root);
     let ancestor = root;
+    // generate related alerts for root
+    const processDuration: number = 6 * 3600;
+    if (this.randomN(100) < pctWithRelated) {
```

Comment:
hmm I think this block needs to be in the ancestry `for` loop right? Otherwise this will only generate a related event for the `root` node, and the most recently created ancestor?

Also should we be using `yield` for these like in other places?

Reply:
it is also in the for loop (see below)

Looks like they were using the `yield*` in the other place because that was a generator. This function isn't iterating anything  back so I don't think we need it here.

Reply:
This block handles the root node, then theres another block in the loop that handles the rest of the ancestors.

I didn't bother changing the ancestry function over to use `yield` since the number of nodes doesn't blow up exponentially the way the children do so returning the whole array of events at the end here is fine imo.

Reply:
It might be nice to move the handling of the root node inside the loop to avoid duplicating the code but I don't think its necessary for this PR.

Reply:
correction for clarity:
Below `*descendantsTreeGenerator` is a generator, so it has to yield all the events back.
In this function `createAlertEventAncestry`, we are just returning an array of events instead of yielding them back.

The approach below is a little more ES6-y, but they both work.

Reply:
Ah thanks, sorry I missed that.